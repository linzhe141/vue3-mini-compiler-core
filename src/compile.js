import { baseParse } from './parse.js'
import { generate } from './codegen.js'
import { transform } from './transform.js'
import { transformElement } from './transforms/transformElement.js'
import { transformText } from './transforms/transformText.js'
import { transformOn } from './transforms/vOn.js'
export function baseCompile(source, options = {}) {
  const mode = options.mode || 'module'

  const ast = baseParse(source)
  transform(ast, {
    nodeTransforms: [transformElement, transformText],
    directiveTransforms: {
      on: transformOn,
    },
  })

  return generate(ast, { mode })
}
