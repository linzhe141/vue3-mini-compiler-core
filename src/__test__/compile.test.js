import { expect, describe, test } from 'vitest'
import { baseCompile as compile } from '../compile'
describe('codegen', () => {
  test('element', () => {
    const input = `<div id="test" disabled x @click="()=>foo()">{{yy}}</div>`
    const { code } = compile(input)
    expect(code).toMatchSnapshot()
  })
})
