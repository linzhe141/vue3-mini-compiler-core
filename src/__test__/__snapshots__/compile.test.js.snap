// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`codegen > element 1`] = `
"import {toDisplayString as _toDisplayString, createText as _createText, createElementVnode as _createElementVnode, openBlock as _openBlock, createElementBlock as _createElementBlock} from "vue"
export function render(_ctx, _cache) {return (_openBlock(), _createElementBlock("div", { id: "test",disabled: "",x: "",onClick: ()=>foo() }, [_createText(_toDisplayString(yy))]))}"
`;
